src/Slices/Dashboard/UI/GraphCard.tsx

```diff
@@ -27,40 +27,38 @@ export const GraphCard: React.FC<GraphCardProps> = ({
 }) => {
   const [chartState, setChartState] = useState<Metric[]>([]);
   const [legendDataState, setLegendDataState] = useState<LegendData[]>([]);
-  const [minMax, setMinMax] = useState<number[]>([]);
+  const [max, setMax] = useState<number>(0);
 
   useEffect(() => {
```

Comment:
A React component is rerendered whenever a prop changes. So your metric prop is the one that can trigger the changes. 
Based on that logic, I'd expect you wouldn't have needed to use State. 

if you got methods to transform your data, before passing them to the Card component, this should work, no ? I get a feeling you get stuck with different states at different levels now, that are interfering with each other. 

`  legendData={legendDataState}` could be `  legendData={formatLegendData()}`

Reply:
I forgot about that, I'll change that